doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+
+In order to give users complete freedom in implementing neuron model dynamics, NESTML has a full procedural programming language built in. This programming language is used to define the model's time update and functions.
+
+### Data types and physical units
+
+Data types define types of variables as well as parameters and return values of functions. NESTML provides the following primitive types and physical data types:
+
+#### Primitive data types
+* `real` corresponds to the `double` data type in C++. Example literals are: `42.0`, `-0.42`, `.44`
+* `integer` corresponds to the `long` data type in C++. Example literals are: `42`, `-7`
+* `boolean` corresponds to the `bool` data type in C++. Its only literals are `true` and `false`
+* `string` corresponds to the `std::string` data type in C++. Example literals are: `"Bob"`, `""`, `"Hello World!"`
+* `void` corresponds to the `void` data type in C++. No literals are possible and this can only be used in the declaration of a function without return value.
+
+#### Physical units
+
+A physical unit in NESTML can be either a simple physical unit or a complex physical unit. A simple physical unit is composed of an optional magnitude prefix and the name of the unit.
+
+The following table lists seven base units, which can be used to specify any physical unit. This idea is based on [the SI units](https://en.wikipedia.org/wiki/International_System_of_Units).
+
+|Quantity | Unit Name | NESTML/SI unit |
+|---|---|---|
+| length              | metre    | m |
+| mass                | kilogram | kg |
+| time                | second   | s |
+| electric current    | ampere   | A |
+| temperature         | kelvin   | K |
+| amount of substance | mole     | mol |
+| luminous intensity  | candela  | cd |
+
+Any other physical unit can be expressed as a combination of these seven units. These other units are called derived units. NESTML provides the following derived units as built-in data types:
+
+|Quantity | Unit name | Definition | NESTML/SI unit |
+|---|---|---|---|
+| capacitance            | farad   | A<sup>2</sup> s<sup>4</sup> kg<sup>−1</sup> m<sup>−2</sup> | F |
+| electrical conductance | siemens | A<sup>2</sup> s<sup>3</sup> kg<sup>−1</sup> m<sup>−2</sup> | S |
+| electric charge        | coulomb | A s                                                        | C |
+| electric potential     | volt    | kg m<sup>2</sup> A<sup>−1</sup> s<sup>−3</sup>             | V |
+| electric resistance    | ohm     | kg m<sup>2</sup> A<sup>−2</sup> s<sup>−3</sup>             | ohm/&‌#x03A9; |
+| frequency              | hertz   | s<sup>−1</sup>                                             | Hz |
+
```

Comment:
@abigailm, @DimitriPlotnikov: I think these units suffice for writing neuron models. I'd rather leave out the ones below. We can always add more if people request them. At least we should not have them in the reference manual.